app/assets/javascripts/components/common/wiki_overview_stats.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import WikiOverviewTabs from './OverviewStats/wiki_overview_tabs';
+import NamespaceOverviewStats from './OverviewStats/namespace_overview_stats';
+import WikidataOverviewStats from './wikidata_overview_stats';
+
+import ArticleUtils from '../../utils/article_utils';
+
+
+const WikiOverviewStats = ({ wikidataStats, wikiNamespaceStats }) => {
+  const [currentTabId, setCurrentTabId] = useState(0);
+
+  const getTabTitle = (id, wiki) => {
+    const project = wiki.split('.')[1];
+    let ns_title = ArticleUtils.NamespaceTitleFromId[id];
+    if (typeof (ns_title) !== 'string') ns_title = ns_title[project];
+    return `${I18n.t(`namespace.${ns_title}`)} (${wiki})`;
+  };
+
+  const getContentTitle = (id, wiki) => {
+    const project = wiki.split('.')[1];
+    let ns_title = ArticleUtils.NamespaceTitleFromId[id];
+    if (typeof (ns_title) !== 'string') ns_title = ns_title[project];
+    return `Wiki: ${wiki}, Namespace: ${I18n.t(`namespace.${ns_title}`)}`;
+  };
+
+  const onTabChange = (e) => {
+    return setCurrentTabId(Number(e.currentTarget.id));
+  };
+
+  const hasWikidataStats = wikidataStats !== null;
+  const isWikidata = (hasWikidataStats && currentTabId === 0);
+
+  let statsData = [];
+  if (hasWikidataStats) {
+    statsData.push({ id: 0, tabTitle: 'Wikidata', data: wikidataStats });
+  }
+
+  let i = hasWikidataStats ? 1 : 0;
```

Comment:
I guess `i` here is an index for which tab of stats it is? This would benefit from a more meaningful variable name, or a comment explaining how this iterator variable should be understood.